File: shoring/compat/src/main/java/com/panopset/compat/Rezop.java
package com.panopset.compat;

import java.io.File;
import java.io.IOException;
import java.io.InputStream;
import java.io.StringWriter;
import java.util.List;

public class Rezop {
  
  private Rezop() {}

  public static List<String> textStreamToList(InputStream is) {
    String lines = Streamop.getTextFromStream(is);
    return Stringop.stringToList(lines);
  }

  public static String getPackageResourcePath(Package pkg) {
    StringWriter sw = new StringWriter();
    sw.append("/");
    sw.append(pkg2path(pkg.getName()));
    return sw.toString();
  }

  public static String pkg2path(String dotName) {
    return dotName.replace(".", "/");
  }

  public static void copyTextResourceToFile(Class<?> clazz, String resourcePath, File file)
      throws IOException {
    Fileop.copyInputStreamToFile(clazz.getResourceAsStream(resourcePath), file);
  }

  public static void copyTextResourceToFile(Class<?> clazz, String resourcePath, String fileName)
      throws IOException {
    Fileop.copyInputStreamToFile(clazz.getResourceAsStream(resourcePath), fileName);
  }

  public static String pkg2resourcePath(Class<?> clazz) {
    StringWriter sw = new StringWriter();
    sw.append("/");
    sw.append(clazz.getPackage().getName().replace(".", "/"));
    sw.append("/");
    return sw.toString();
  }
}

File: code/shoring/compat/src/main/kotlin/com/panopset/compat/Rezop.kt
package com.panopset.compat

import com.panopset.compat.Fileop.copyInputStreamToFile
import com.panopset.compat.Streamop.getTextFromStream
import com.panopset.compat.Stringop.stringToList
import java.io.File
import java.io.InputStream
import java.io.StringWriter

object Rezop {
    fun textStreamToList(inputStream: InputStream?): List<String> {
        if (inputStream == null) {
            return ArrayList()
        }
        val lines = getTextFromStream( inputStream)
        return stringToList( lines)
    }

    fun getPackageResourcePath(pkg: Package): String {
        val sw = StringWriter()
        sw.append("/")
        sw.append(pkg2path(pkg.name))
        return sw.toString()
    }

    fun pkg2path(dotName: String): String {
        return dotName.replace(".", "/")
    }

    fun copyTextResourceToFile(clazz: Class<*>, resourcePath: String, file: File) {
        val inputStream = clazz.getResourceAsStream(resourcePath)
        if (inputStream == null) {
            Logz.warn("$resourcePath not found.")
        } else {
            copyInputStreamToFile(inputStream, file)
        }
    }

    fun copyTextResourceToFile(clazz: Class<*>, resourcePath: String, fileName: String) {
        val inputStream = clazz.getResourceAsStream(resourcePath)
        if (inputStream == null) {
            Logz.warn("$resourcePath not found.")
        } else {
            copyInputStreamToFile(inputStream, fileName)
        }

    }

    fun pkg2resourcePath(clazz: Class<*>): String {
        val sw = StringWriter()
        sw.append("/")
        sw.append(clazz.getPackage().name.replace(".", "/"))
        sw.append("/")
        return sw.toString()
    }
}
